kpi/utils/xml.py

```diff
@@ -154,3 +155,37 @@ def add_xml_declaration(xml_content: Union[str, bytes]) -> Union[str, bytes]:
     if use_bytes:
         return xml_.encode()
     return xml_
+
+
+def edit_submission_xml(
+    xml_parsed: etree._Element,
+    path: str,
+    value: str,
+) -> None:
+    """
+    Edit submission XML with an XPath and new value, creating a new tree
+    element if the path doesn't yet exist.
+    """
+    if '/' in path:
+        accumulated_elements = []
+        for i, element in enumerate(path.split('/')):
+            if i == 0:
+                if xml_parsed.find(element) is None:
+                    etree.SubElement(xml_parsed, element)
+                accumulated_elements.append(element)
```

Comment:
Suggestion: use a slice instead of this extra logic and the `accumulated_elements` list.
```
In [1]: path = 'a/b/c/d'

In [2]: path_parts = path.split('/')

In [3]: for i, el in enumerate(path_parts):
   ...:     print(path_parts[:i+1])
   ...: 
['a']
['a', 'b']
['a', 'b', 'c']
['a', 'b', 'c', 'd']
```